cluster-autoscaler/config/autoscaling_options.go

```diff
@@ -121,4 +121,7 @@ type AutoscalingOptions struct {
 	Regional bool
 	// Pods newer than this will not be considered as unschedulable for scale-up.
 	NewPodScaleUpDelay time.Duration
+	// Exclude pods that are currently not scheduled, but, can be scheduled on current empty cluster capacity during scale up.
+	// Pods with nominatedNodeName set are always excluded
```

Comment:
1) I just wanted to add the information that pods with nominatedNodeName are always excluded for CA scale up.

2) Will do

Reply:
Since the code for excluding pods with nominatedNodeName is already present before this PR, I'll only be adding the packing based filtering as a commit.

Reply:
The code you pointed just splits list of pending pods into those with nominated-node-name and those without.
What is missing is actually simulating scheduling of those pods with nominated-node-name to the node which is nominated for them. You need to add logic for that to `FilterOutSchedulable`, which takes the list of those pods via `podsWaitingForLowerPriorityPreemption` argument.

Reply:
... Or I am wrong.
Did  you mean that `CreateNodeNameToInfoMap` called in `FilterOutSchedulable` already does the job of simulating scheduling?

Reply:
Yes. I seemed to have accidentally removed the logic in my last commit. Will add it back.

Reply:
Sorry for the confusion. The pods in `podsWaitingForLowerPriorityPreemption` are already "scheduled" on the nominated node when we call `scheduler_util.CreateNodeNameToInfoMap` (https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/core/utils.go#L116)

Reply:
Yeah - now I understand (btw. thanks @aleksandra-malinowska for help on that). It appears that we have code for that already. Yet the `CreateNodeNameToInfoMap` is broken. It uses `NominatedNodeName` annotation, which was right when priority and preemption was still in alpha. Later the nominated-node-name migrated to be an API field, and `pod.Status.InitContainerStatuses` should be used. Can you please fix that?

Reply:
Yes. Sure!